common/src/sql/dbinit.sql

```diff
@@ -107,14 +107,10 @@ CREATE TABLE omicron.public.sled (
 );
 
 /* Add an index which lets us look up sleds on a rack */
-CREATE INDEX ON omicron.public.sled (
-    rack_id
-) WHERE time_deleted IS NULL;
-
-CREATE INDEX ON omicron.public.sled (
```

Comment:
This index is superfluous because [CockroachDB always has an index on the primary key](https://www.cockroachlabs.com/docs/stable/primary-key.html#details).  (Note that having an index on "rack_id" and one on "id" is not the same as having one on `(rack_id, id)`.)

Reply:
This index also includes a filter predicate `time_deleted IS NULL` which makes it a partial index no? Perhaps not super relevant for this particular table but in the limit there could be more old/deleted records that a partial index could speed up access for only live records.

Reply:
Thanks for pointing that out.  I had missed that.  Are you saying that even if CockroachDB already indexes on "id", it might be beneficial to have an index on "id" limited only to live rows because querying it could be faster?  Yeah, tThat's definitely possible.  I think we probably want to wait and see if that's true, though, since it's somewhat speculative and has downsides (i.e., the extra disk space and memory usage and cost to update it on writes).